drivers/spi/spi_nrfx_spim.c

```diff
@@ -627,15 +652,18 @@ static int spi_nrfx_init(const struct device *dev)
 		},							       \
 		.irq_connect = irq_connect##idx,			       \
 		.pcfg = PINCTRL_DT_DEV_CONFIG_GET(SPIM(idx)),		       \
-		.max_chunk_len = BIT_MASK(SPIM_PROP(idx, easydma_maxcnt_bits)),\
+		.max_chunk_len = SPI_NRFX_SPIM_MAX_CHUNK_LEN(idx),	       \
```

Comment:
This should not be changed. When the buffer supplied in a request is DMA-accessible, then the internal buffer is not used for the transaction and its size should not limit the chunk length. Thus, `max_chunk_len` should be based on the hardware limit as it was so far.
For the case when the internal buffer is used, the chunk length should be limited at runtime by
```
if (chunk_len > CONFIG_SPI_NRFX_RAM_BUFFER_SIZE) {
	chunk_len = CONFIG_SPI_NRFX_RAM_BUFFER_SIZE;
}
```